cmd/skopeo/inspect.go

```diff
@@ -185,3 +187,15 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 	fmt.Fprintf(stdout, "%s\n", string(out))
 	return nil
 }
+
+func retryGetManifest(ctx context.Context, instanceDigest *digest.Digest, maxRetry int, retryDelay time.Duration, src types.ImageSource) ([]byte, error) {
+	rawManifest, _, err := src.GetManifest(ctx, nil)
+	for retries := 0; err != nil && isRetryable(err) && retries < maxRetry; retries++ {
+		logrus.Infof("Warning: failed, retrying in %s seconds... (%d/%d)", retryDelay, retries+1, maxRetry)
+		time.Sleep(retryDelay)
```

Comment:
This should notice cancelation of `ctx` so that `--command-timeout` still applies.

Reply:
move context timeout error check to isRetryable

Reply:
Marking timeout errors as not retrial is useful, but that’s not quite the same thing:

Assume `--initial-retry-delay` is 5 (maybe by default) and `--command-timeout` is 3. In that case this `time.Sleep` should terminate after 3 seconds, not 5 (or maybe it should not even be attempted and the command should fail with a timeout early).

See the example in https://golang.org/pkg/time/#After ; this can similarly combine `time.After` and `ctx.Done()`.